### PR TITLE
Projects list

### PR DESCRIPTION
**Pull request by srobert99**
#### Description
Added projects-list screen (data got from backend)

#### [Task](task_url)
https://trello.com/c/UvLfqW6k/171-implement-projects-screen

### CODE REVIEW

**Comment by srobert99** [comment]
[Implement projects screen](https://trello.com/c/UvLfqW6k/171-implement-projects-screen)

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <resources>
-    <string name="app_name">Freaks Catalog</string>
+    <string name="app_name">Freaks Robinho</string>
```
Comment (cipry13) [inline app/src/main/res/values/strings.xml]: You should remove testing code once you're done testing

File: buildSrc/src/main/kotlin/BuildAndroidConfig.kt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 object BuildAndroidConfig {
-    const val APPLICATION_ID = "com.agilefreaks.freaks_catalog"
+    const val APPLICATION_ID = "com.agilefreaks.freaks.catalog.robinho"
```
Comment (cipry13) [inline buildSrc/src/main/kotlin/BuildAndroidConfig.kt]: You should remove testing code once you're done testing

**Comment by cipry13** [comment]
`FreakDetailsRepositoryImplTest.kt` and `ProjectsViewModelTest.kt` are not projects list related

**Comment by deniscoman** [comment]
Base for this PR should be project-screen instead of develop

File: features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        return ItemViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
+        val project = projectsList[position]
+        holder.bind(project)
+    }
+
+    override fun getItemCount(): Int = projectsList.size
+
```
Comment (deniscoman) [inline features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectAdapter.kt:23]: Remove extra empty line

File: features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectsFragment.kt [hunk 4]
```diff
@@ -6,19 +6,30 @@ import android.view.View
...
         viewBinding =
             DataBindingUtil.inflate(inflater, R.layout.fragment_projects, container, false)
         viewBinding.lifecycleOwner = viewLifecycleOwner
+        viewBinding.projectsViewModel = viewModel
+
+        val recyclerView = viewBinding.projectsRecyclerview
+        recyclerView.layoutManager = LinearLayoutManager(context)
+        viewModel.projects.observe(viewLifecycleOwner, {projects ->
+            recyclerView.adapter = ProjectAdapter(projects)
+        })
```
Comment (deniscoman) [inline features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectsFragment.kt]: We can extract this in a method called something like "setupRecycleView"

**Review by deniscoman — CHANGES_REQUESTED** [review]

File: features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectsModule.kt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.agilefreaks.freaks_catalog.features.projects
+
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+
```
Comment (cipry13) [inline features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectsModule.kt]: remove extra empty line

File: features/projects/src/main/res/layout/fragment_projects.xml [hunk 6]
```diff
@@ -2,13 +2,39 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="@+id/projects_recyclerview"
+            app:layout_constraintStart_toStartOf="parent"
+            android:visibility="@{projectsViewModel.isLoading() ? View.VISIBLE : View.GONE}"
+            app:layout_constraintTop_toTopOf="parent" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+
```
Comment (cipry13) [inline features/projects/src/main/res/layout/fragment_projects.xml]: remove extra empty line

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/FreakDetailsDataSource.kt [hunk 7]
```diff
@@ -5,7 +5,7 @@ import com.apollographql.apollo.coroutines.await
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
-interface FreakDetailsDataSource {
+interface   FreakDetailsDataSource {
```
Comment (DeeaMatei) [inline features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/FreakDetailsDataSource.kt]: Remove extra white space

**Review by DeeaMatei — CHANGES_REQUESTED** [review]

**Review by DeeaMatei — APPROVED** [review]
Looks good :octocat:

**Review by AlexandraDamaschin — APPROVED** [review]
Does this improve software quality? Yes

**Pull request merged**